libdrgn/orc_info.c

```diff
@@ -367,18 +526,40 @@ struct drgn_error *drgn_module_parse_orc(struct drgn_module *module)
 {
 	struct drgn_error *err;
 
-	if (module->debug_file->platform.arch->arch != DRGN_ARCH_X86_64)
+	if (module->prog->platform.arch->arch != DRGN_ARCH_X86_64)
 		return NULL;
 
-	// pc_offsets and entries point to the Elf_Data buffers until we're
-	// done. We don't want those freed by drgn_module_orc_info_deinit(), so
-	// clear them if anything goes wrong.
+	// Depending on the source of pc_offsets and entries, we may or may not
+	// need to free them. We should always clear the pointers out of the
+	// module to avoid an invalid or double free in
+	// drgn_module_orc_info_deinit().
```

Comment:
Could you please retain a mention of how the buffers temporarily come from `Elf_Data` in the not built-in case?